src/hardware/pounder/mod.rs

```diff
@@ -91,48 +93,87 @@ impl From<Channel> for GpioPin {
     }
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct DdsChannelState {
-    pub phase_offset: f32,
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, Miniconf)]
+pub struct DdsChannelConfig {
     pub frequency: f32,
+    pub phase_offset: f32,
     pub amplitude: f32,
-    pub enabled: bool,
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct ChannelState {
-    pub parameters: DdsChannelState,
-    pub attenuation: f32,
+impl Default for DdsChannelConfig {
+    fn default() -> Self {
+        Self {
+            frequency: 10.0e6,
+            phase_offset: 0.0,
+            amplitude: 0.0,
+        }
+    }
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct InputChannelState {
-    pub attenuation: f32,
-    pub power: f32,
-    pub mixer: DdsChannelState,
+impl DdsChannelConfig {
+    pub fn try_into_dds_profile(
+        self,
+        system_clock_frequency: f32,
+    ) -> Result<DdsProfile, ad9959::Error> {
+        DdsProfile::new(
+            self.frequency,
+            self.phase_offset,
+            self.amplitude,
+            system_clock_frequency,
+        )
+    }
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct OutputChannelState {
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, Miniconf)]
+pub struct ChannelConfig {
+    pub dds: DdsChannelConfig,
     pub attenuation: f32,
-    pub channel: DdsChannelState,
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
+impl Default for ChannelConfig {
+    fn default() -> Self {
+        ChannelConfig {
+            dds: DdsChannelConfig::default(),
+            attenuation: 31.5,
+        }
+    }
+}
+
+#[derive(
+    Serialize, Deserialize, Copy, Clone, Debug, PartialEq, MiniconfAtomic,
+)]
 pub struct DdsClockConfig {
     pub multiplier: u8,
     pub reference_clock: f32,
     pub external_clock: bool,
 }
 
+impl Default for DdsClockConfig {
+    fn default() -> Self {
+        Self {
+            multiplier: super::design_parameters::DDS_MULTIPLIER,
```

Comment:
If you make this configurable, then these are defnitely not "design parameters" anymore. You need to remove them if someone relies on them being constants.